### ISSUE TITLE
MSVC: string literal defines in rsp files can be processed incorrectly

### ISSUE THREAD

**Issue by Predelnik** [ccache/ccache#1072]
### How to reproduce ###

`a.cpp:`
```cpp
const char *s = "abc" MY_DEFINE "def";
```
`a.rsp`
```bash
"-DMY_DEFINE=\"some_value\""
```
Executing 

1.
```bash
cl /c a.cpp @a.rsp
```

2.
```bash
ccache cl /c a.cpp @a.rsp
```

### Actual behavior ###

1. Compiles without error
2. Result in an error:
```
a.cpp
a.cpp(1): error C2017: illegal escape sequence
a.cpp(1): error C2146: syntax error: missing ';' before identifier 'some_value'
a.cpp(1): error C2143: syntax error: missing ';' before 'string'
a.cpp(1): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
```

### Expected behavior ###
Both compile without error

### Environment ###
Windows 10, Visual Studio 2022

ccache version 4.6

**Comment by jrosdahl** [comment]
I believe this is fixed by #1071.

**Comment by Predelnik** [comment]
Ah, sorry, didn't check master version. Thank you.
